servers/xgboostserver/xgboostserver/XGBoostServer.py

```diff
@@ -22,7 +25,27 @@ def load(self):
         model_file = os.path.join(
             seldon_core.Storage.download(self.model_uri), BOOSTER_FILE
         )
-        self._booster = xgb.Booster(model_file=model_file)
+        if not os.path.exists(model_file):
+            # Fallback to deprecated .bst format
+            model_file = os.path.join(
+                seldon_core.Storage.download(self.model_uri), BOOSTER_FILE_DEPRECATED
+            )
+            if os.path.exists(model_file):
+                logger.warning(
+                    "Using deprecated .bst format for XGBoost model. "
+                    "Please update to the .json format in the future."
+                )
+            else:
+                raise FileNotFoundError(f"Model file not found: {BOOSTER_FILE} or {BOOSTER_FILE_DEPRECATED}")
+
+        if version.parse(xgb.__version__) < version.parse("1.7.0"):
```

Comment:
is this code path possible given that we build the server with a specific `xgboost >= 1.7.0, <=2.2.0`?